packages/mui-base/src/PopperUnstyled/PopperUnstyled.d.ts

```diff
@@ -14,6 +14,11 @@ export interface PopperUnstyledProps
    * The return value will passed as the reference object of the Popper instance.
    */
   anchorEl?: null | VirtualElement | (() => VirtualElement);
+  /**
+   * The component used for the root node.
+   * Either a string to use a HTML element or a component.
+   */
+  component?: React.ElementType;
```

Comment:
I'm OK with adding the `component` prop. However let's make the Popper an `OverridableComponent` and make it consistent with the others. This means defining a TypeMap interface and adding a `components` prop as well. A standard test to verify if the `component` prop works well could also be added. See #33506 for all the changes needed to correctly implement an overridable component.